src/circuit/mod.rs

```diff
@@ -1301,9 +1306,27 @@ mod test {
         (verification_key, proof)
     }
 
+    fn check_if_crs_exists_and_download(n: usize) {
```

Comment:
I would return the file path (and change the name to the function). 

So I would have 
```rust
let file_path = std::path::Path::new(file_path);
if file_path.exists() {
  println!();
} else {
  // downloading file
} 

return file_path
```